### ISSUE TITLE
[Bug]: Input controls' EditArea focus colouring is not themed

### ISSUE THREAD

**Issue by lesandrog** [Krypton-Suite/Standard-Toolkit#1826]
The Krypton ComboBox component with the DropDownStyle property = ComboBoxStyle.DropDownList, when clicked or given focus, does not display any visual selection feature.

I think this makes it difficult to navigate using keys (TAB).

The default Windowsform combobox and the ComponentFactory have different behavior.

This is a sample of the behaviour that needs correcting.
![image](https://github.com/user-attachments/assets/26071de5-f6cd-41b0-971e-953c0dd7baad)

## Controls affected:
- `KComboBox`
- `KRibbonComboBox` (needs a check)
- `KDateTimePicker`
- `KDomainUpDown`
- `KListView` (all view modes affected)
- `KMaskedTextBox`
- `KNumericUpdown`
- `KTextBox`

Other controls that are derived from the above might override behaviour and need to have a check when the above have been corrected.

Sample project:
[WindowsApp3.zip](https://github.com/user-attachments/files/17481540/WindowsApp3.zip)

**Comment by giduac** [comment]
Hi @lesandrog, @PWagner1, @Smurf-IV,

I've been looking into this but the default behaviour is overridden by the theming. When the combobox has the focus and you cycle through the items by keyboard you can see the inner combobox displaying this blue/white colouring. Right after that the theming kicks in and repaints it.

The question here if this is something we want to change or the newer way this functions is the way to go?

Your input please...

**Comment by Smurf-IV** [comment]
The ComboBox has a Theming Focus style, so things like the border colour can be changed ( thickness / rounding etc)
As for making the edit section have a colour change (I though that was also in the themeing focus section ?)
Anyway, this is obviously a regression as shown by the OP.

**Comment by giduac** [comment]
@Smurf-IV 
What is the Theming focus section and where to find it?

**Comment by Smurf-IV** [comment]
Added `themes`, as V90 was supposed to be themed fixed !
Removed V85 as there is a workaround

**Comment by giduac** [comment]
@Smurf-IV 
I actually tried StateActive, but it stays active all the time not only when the control has focus / the cursor is in the edit area.

Back to the default from the inner combo isn't really an option since that won't always have the look you want with each theme on it's own.

**Comment by Smurf-IV** [comment]
> @Smurf-IV I actually tried StateActive, but it stays active all the time not only when the control has focus / the cursor is in the edit area.


You have to ensure that the following is set to false:
![image](https://github.com/user-attachments/assets/be724e57-e3dc-4be5-8988-ba718b78bd60)

**Comment by Smurf-IV** [comment]
> Back to the default from the inner combo isn't really an option since that won't always have the look you want with each theme on it's own.

I do not understand ?

**Comment by giduac** [comment]
@Smurf-IV 
Yes, that works.
But that would make it an option the user can configure...imo...?
At least the themes don't configure it and as it looks like those use/inherit the colors the StateCommon/Normal is using??

**Comment by giduac** [comment]
@lesandrog 

Can you have look at the above and give it a try to configure this manually.
Please let us know what you think of it...

**Comment by lesandrog** [comment]
@giduac 

For now I'll use it this way, Color of StateActive/ComboBox/Back: 0;120;215 and AlwaysActive: False.

The downsides are:
When you hover over the combobox, the color changes, even before you click.
If you use a theme control, when you change to a dark theme, this component will remain blue.
And you need to manually configure all the comboboxes.

I don't think this is the most appropriate or practical solution.
I hope friends can develop something more practical.

Thank you, and congratulations on your work.

**Comment by Smurf-IV** [comment]
> On the other hand the questions remains if it should be part of the theming and made an options that can toggled.

It should be a theme thing, and the default should be made `AllwaysActive = False` (Breaking Changes!)
Not sure about toggling, as that is what the ability to override `StateActive` is all about.

OR:
Investigate why the focus colour (Which will not change with the theme!) is not used any more ! (But then that goes against using a theme for focus etc!)

**Comment by giduac** [comment]
> > On the other hand the questions remains if it should be part of the theming and made an options that can toggled.
> 
> It should be a theme thing, and the default should be made `AllwaysActive = False` (Breaking Changes!) Not sure about toggling, as that is what the ability to override `StateActive` is all about.
> 
> OR: Investigate why the focus colour (Which will not change with the theme!) is not used any more ! (But then that goes against using a theme for focus etc!)

@Smurf-IV 
cc: @lesandrog (if you have ideas/thoughts don't hesitate ... )

Yes, AlwaysActive can be or rather is already the toggle for this.
The focus fore- and background colour should be defined by or derived from the theme.

Is this one that should go into the KCustomePaletteBase ***only*** as we are switching to embedded XML themes in V100.... ?
Should it, for that reason, be kept out of V95. This was the first ticket on this and that'll save us a lot of work on V95....?

**Comment by lesandrog** [comment]
I've been looking at textboxes, comboboxes (dropdowns) and other components and the text selection is always blue, no matter the theme.

This focus color seems to be stuck.

In my opinion, it could be stuck, unless in the future it will be possible to customize the focus color on all components.
![image](https://github.com/user-attachments/assets/5ae85b20-b23b-4524-b999-02c77151044c)

**Comment by giduac** [comment]
@lesandrog 

Could you have a look at DomainUpDown and NumericUpdown ?

**Comment by lesandrog** [comment]
@giduac 
For these components, the text selection is also blue.

![image](https://github.com/user-attachments/assets/43720659-1a26-49bb-98e2-f8eb785f62c4)

So this concern could be eliminated:
>  If you use a theme control, when you change to a dark theme, this component will remain blue.

**Comment by Smurf-IV** [comment]
> If you use a theme control, when you change to a dark theme, this component will remain blue.

I _think_ that colour comes from the current active Windows Theme.

**Comment by Smurf-IV** [comment]
> Is this one that should go into the KCustomePaletteBase only as we are switching to embedded XML themes in V100.... ?
> Should it, for that reason, be kept out of V95. This was the first ticket on this and that'll save us a lot of work on V95....?

No, AS this is a regression away from what used to happen (i.e. no highlighting at all in V85 onwards !)
So, it will Need to go into V95 as V90 was supposed to be the fixing of all things "themey" (Is that a word.. It is now :-))

**Comment by giduac** [comment]
Hi @Smurf-IV, @PWagner1 & @lesandrog,

I've updated the topic start and added a list with all controls affected.
